internal/client_test.go

```diff
@@ -0,0 +1,61 @@
+package client_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	client "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/testutil"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_OnAfterResponseErrMiddleware(t *testing.T) {
```

Comment:
I'm not sure if we shouldn't have such a test for each method that is calling the API 🤷  instead of single one for example method.